src/NuGet.Core/NuGet.Frameworks/CompatibilityProvider.cs

```diff
@@ -48,11 +48,13 @@ public bool IsCompatible(NuGetFramework target, NuGetFramework candidate)
             // check the cache for a solution
             var cacheKey = new CompatibilityCacheKey(target, candidate);
 
-            bool? result = _cache.GetOrAdd(cacheKey, (Func<CompatibilityCacheKey, bool>)((key)
-                =>
-            { return IsCompatibleCore(target, candidate) == true; }));
+            if (!_cache.TryGetValue(cacheKey, out bool result))
+            {
+                result = IsCompatibleCore(target, candidate) == true;
+                _cache.TryAdd(cacheKey, result);
```

Comment:
```suggestion
                _cache.TryAdd(cacheKey, IsCompatibleCore(target, candidate));
```

Reply:
@aortiz-msft: The suggested fix does not set the result in the case where the cache did not have the value. We need to store the result of IsCompatibleCore in the result because the method returns that result in the end.

Reply:
Furthermore, when debugging, it's nicer to have each expression/method result assigned to different variables, so I can easily see the value that `IsCompatibleCore` returns without stepping into it.  The JIT compiler and/or release binary will optimize the variable away, so it doesn't have runtime perf issues.